config/gulp/file-process.js

```diff
@@ -1,280 +1,137 @@
-                },
-              },
-              {
-                width: 2400,
-                rename: {
-                  suffix: "_w2400",
-                },
-              },
-              {
-                width: 2400,
-                rename: {
-                  suffix: "_w2400",
-                  extname: ".webp",
-                },
-              },
-              {
-                width: 2400,
-                grayscale: true,
-                rename: {
-                  suffix: "_w2400bw",
-                },
-              },
-              {
-                width: 2400,
-                grayscale: true,
-                rename: {
-                  suffix: "_w2400bw",
-                  extname: ".webp",
-                },
-              },
-              {
-                // original -> grayscale
-                grayscale: true,
-                rename: {
-                  suffix: "_bw",
-                },
-              },
-              {
-                // original -> grayscale webp
-                grayscale: true,
-                rename: {
-                  suffix: "_bw",
-                  extname: ".webp",
-                },
-              },
-              {
-                // original -> webp
-                rename: {
-                  suffix: "",
-                  extname: ".webp",
-                },
-              },
-              {
-                // Empty case to produce a copy of the original
-              },
-            ],
-          },
-          {
-            // Global configuration for all images
-            quality: 80,
-            compressionLevel: 7,
-            progressive: true,
-            withMetadata: false,
-            errorOnUnusedConfig: false,
-            skipOnEnlargement: true,
-            errorOnEnlargement: false,
-            silent: true,
-          }
-        )
-      )
-      .pipe(dest("content/uploads/_working-images/processed/"))
+const processImagesDirectory = "./content/uploads/_working-images/to-process";
+const processedImagesDirectory = "./content/uploads/_working-images/processed";
+
+/**
+ * Settings object for each image variant width and suffix
+ * Device name is not used in image name
+ */
+const variantSettings = {
+  mobile: {
+    width: 200,
+    suffix: "_w200",
+  },
+  tablet: {
+    width: 400,
+    suffix: "_w400",
+  },
+  desktop: {
+    width: 800,
+    suffix: "_w800",
+  },
+};
+
+/**
+ * Separate image name, extension and directory path and return as an object
+ * @param {string} image - original image filename
+ */
+
+function getImageDetails(image) {
+  const [imageName, imageExtension] = image.split(".");
+  // get the full filepath for the provided image
+  const imagePath = path.join(
+    path.resolve(),
+    `${processImagesDirectory}`,
+    image
+  );
+
+  const imageVariant = {
+    name: imageName,
+    extension: imageExtension,
+    path: imagePath,
+  };
+  return imageVariant;
+}
```

Comment:
## Optional

You could leverage [Path.parse()](https://nodejs.org/api/path.html#pathparsepath) here to use the native method for getting this information.